src/logic/render/PolygonRenderEngine.ts

```diff
@@ -15,36 +40,329 @@ export class PolygonRenderEngine extends BaseRenderEngine {
+                return index === this.resizedAnchorIndex ? snappedMousePosition :
+                    PointUtil.add(PointUtil.multiply(point, 1/data.activeImageScale), data.activeImageRectOnCanvas);
+            });
+            this.drawPolygon(polygonOnCanvas, true);
+        }
+    }
+
+    private drawExistingLabels(data: EditorData) {
+        const activeLabelId: string = EditorSelector.getActiveLabelId();
+        const highlightedLabelId: string = EditorSelector.getHighlightedLabelId();
+        const imageData: ImageData = EditorSelector.getActiveImageData();
+        imageData.labelPolygons.forEach((labelPolygon: LabelPolygon) => {
+            const isActive: boolean = labelPolygon.id === activeLabelId || labelPolygon.id === highlightedLabelId;
+            const polygonOnCanvas: IPoint[] = labelPolygon.vertices.map((point: IPoint) => {
+                return PointUtil.add(PointUtil.multiply(point, 1/data.activeImageScale), data.activeImageRectOnCanvas);
+            });
+            if (!(isActive && this.isResizeInProgress())) {
+                this.drawPolygon(polygonOnCanvas, isActive);
+            }
+        });
+    }
+
+    private drawPolygon(polygon: IPoint[], isActive: boolean) {
+        const color: string = isActive ? this.config.lineActiveColor : this.config.lineInactiveColor;
+        const standardizedPoints: IPoint[] = polygon.map((point: IPoint) => DrawUtil.setPointBetweenPixels(point));
+        if (isActive) {
+            DrawUtil.drawPolygonWithFill(this.canvas, standardizedPoints, DrawUtil.hexToRGB(color, 0.2));
+        }
+        DrawUtil.drawPolygon(this.canvas, standardizedPoints, color, this.config.lineThickness);
+        if (isActive) {
+            this.mapPointsToAnchors(standardizedPoints).forEach((handleRect: IRect) => {
+                DrawUtil.drawRectWithFill(this.canvas, handleRect, this.config.activeAnchorColor);
+            })
+        }
+    }
+
+    // =================================================================================================================
+    // CREATION
+    // =================================================================================================================
+
+    private updateActivelyCreatedLabel(data: EditorData) {
+        if (this.isCreationInProgress()) {
+            const mousePositionSnapped: IPoint = RectUtil.snapPointToRect(data.mousePositionOnCanvas, data.activeImageRectOnCanvas);
+            this.activePath.push(mousePositionSnapped);
+        } else {
+            const isMouseOverImage: boolean = RectUtil.isPointInside(data.activeImageRectOnCanvas,
+                data.mousePositionOnCanvas);
+            if (isMouseOverImage) {
+                this.activePath.push(data.mousePositionOnCanvas);
+                store.dispatch(updateActiveLabelId(null));
+            }
+        }
+    }
+
+    private cancelLabelCreation() {
```

Comment:
chodziło mi o to żeby było jedno :D w takim razie usuń "cancelLabelCreation". zostaw "finish..." i użyj wszędzie "finish..." :)